### PR TITLE
 Prevent debugger from landing on pseudo-nodes with no id 

### PR DESCRIPTION
**Pull request by haltman-at**
This PR fixes a crash that could occur when debugging contracts with inline assembly.  One of the attributes of an `InlineAssembly` node is `externalReferences`, and nested inside `externalReferences` one can find certain objects I'm calling "pseudo-nodes".  They're not actual nodes -- they lack an `id` and a `nodeType`.  But they do have a `src`, which could cause the debugger to land on them.  The lack of `id` would then cause a crash.

This PR fixes the problem by excluding such pseudo-nodes from the set of nodes the debugger can land on, by only putting nodes with actual IDs into the `IntervalTree`.  This means that instead of landing on these pseudo-nodes, the debugger will land on the `InlineAssembly` node they're a part of.

Note that I'm assuming that this is the only way to get "pseudo-nodes".  If it's not, this might not be the correct solution for those other cases.  But it should be fine here.

### CODE REVIEW

**Comment by haltman-at** [comment]
Interesting question.  We probably could, but I think I'd prefer not to; that's a bit more complex than I'd like to be doing here, I think.

**Comment by haltman-at** [comment]
Actually, thinking about it further, we definitely *don't* want to do that.  Skipped nodes are ones skipped by the interface, but they're often still quite necessary to pass through internally.  Like, variable declarations are skipped, but obviously we need the debugger to pass through those!  Also, note that while I described this change as preventing the debugger from landing on these nodes, it also prevents it from considering one as the *next* node when that matters (that was actually the immediate cause of the crash).  We definitely wouldn't want skipped nodes being skipped for those purposes!

So in short, no, we can't do that here.

**Comment by gnidan** [comment]
Ah of course, duh!
